src/Nethermind/Nethermind.Taiko/TaikoPlugin.cs

```diff
@@ -166,8 +166,8 @@ public async Task InitRpcModules()
                 new BlockInvalidTxExecutor(new BuildUpTransactionProcessorAdapter(scope.TransactionProcessor), scope.WorldState),
                 scope.WorldState,
                 _api.ReceiptStorage,
-                _api.TransactionProcessor,
-                new BeaconBlockRootHandler(_api.TransactionProcessor, _api.WorldStateManager.GlobalWorldState),
+                scope.TransactionProcessor,
```

Comment:
Which one is this? Its used by `TaikoPreparationService`. There is an explicitly created read only env also, but that is not used. The payload preparation service uses block processor instead of block builder?

Reply:
Looks like it should be writable here, because blockProcessor is used blockchainProcessor, maybe we need to change BlockhashStore/BlockProductionWithdrawalProcessor to writable too

Reply:
Ok then, I don't know why it is using the main state, but as long as it is intended then its fine.

Reply:
Or maybe you are right, it does not replace main blockchain processor, probably that scope is needed everywhere

Reply:
I'm not sure I follow. I'll just revert this part of the code and file an issue.